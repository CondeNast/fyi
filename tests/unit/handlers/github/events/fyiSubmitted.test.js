const fyiSubmittedEvent = require('../../../fixtures/fyi-submitted')
const models = require('../../../../../src/models')
const Fyi = models.Fyi

let app
let github

describe('Arch Bot', () => {
  beforeEach(() => {
    ({app, github} = require('../../../setup.js'))
  })

  describe('FYI Submitted', () => {
    it('Creates a Comment in Admin Repository', async () => {
      await app.receive(fyiSubmittedEvent)
      expect(github.issues.removeLabel).toMatchSnapshot()
      expect(github.issues.addLabels).toMatchSnapshot()
      expect(github.issues.createComment).toMatchSnapshot()
      expect(Fyi.forName).toMatchSnapshot()
    })
  })

  afterAll(async () => {
    await models.sequelize.close()
  })
})
