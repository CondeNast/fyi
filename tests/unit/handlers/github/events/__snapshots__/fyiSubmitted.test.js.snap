// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arch Bot FYI Submitted Creates a Comment in Admin Repository 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "name": "fyi-requested",
        "number": 282,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Submitted Creates a Comment in Admin Repository 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "labels": Array [
          "fyi-submitted",
        ],
        "number": 282,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Submitted Creates a Comment in Admin Repository 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "body": "[FYI](https://cnissues.atlassian.net/wiki/spaces/ARCH/pages/357302331/congenial-system) has been submitted.",
        "number": 282,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Submitted Creates a Comment in Admin Repository 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "turbo-octo-engine",
    ],
  ],
}
`;
